torch/utils/bundled_inputs.py

```diff
@@ -0,0 +1,148 @@
+#!/usr/bin/env python3
+from typing import Any, TypeVar, Optional, Tuple, List
+import textwrap
+import torch
+
+
+T = TypeVar("T")
+
+MAX_RAW_TENSOR_SIZE = 16
+
+
+def augment_model_with_bundled_inputs(
+        model: torch.jit.ScriptModule,
+        inputs: Optional[List[Tuple[Any, ...]]] = None,
+        _receive_inflate_expr: Optional[List[str]] = None,  # For debugging.
+        ) -> None:
+    """Add bundled sample inputs to a model.
+
+    Models with bundled inputs can be invoked in a uniform manner by
+    benchmarking and code coverage tools.
+
+    Augmented models will support the following methods:
+
+      `get_all_bundled_inputs() -> List[Tuple[Any, ...]]`
+        Returns a list of tuples suitable for passing to the model like
+        `for inp in model.get_all_bundled_inputs(): model(*inp)`
+
+      `get_num_bundled_inputs() -> int`
+        Equivalent to `len(model.get_all_bundled_inputs())`,
+        but slightly easier to call from C++.
+
+      `run_on_bundled_input(idx: int) -> Any`
+        Run the model on bundled input number `idx`
+
+    Inputs can be specified in one of two ways:
+
+      - The model can define `_generate_bundled_inputs`
+        get_all_bundled_inputs will simply call this method
+        and cache the value.
+      - The `inputs` argument to this function can be a list of tuples,
+        of the same form that will be returned by get_all_bundled_inputs.
+        This function will attempt to optimize arguments so that (e.g.)
+        arguments like `torch.zeros(1000)` will be represented compactly.
+        Only top-level arguments will be optimized.
+        Tensors in lists or tuples will not.
+    """
+    if not isinstance(model, torch.jit.ScriptModule):
+        raise Exception("Only ScriptModule is supported.")
+
+    forward_arg_types = [node.type() for node in list(model.graph.param_node().outputs())[1:]]
```

Comment:
I'm not an expert - but another way would be iterating over model's schema. Probably they would give the same results

Reply:
What is the model's schema and how do I iterate over it?

Reply:
```
m = torch.jit.script(torch.nn.Linear(10,10))
for arg in m.forward.schema.arguments:
  print(arg.name, arg.type)

> self __torch__.torch.nn.modules.linear.Linear
> input Tensor
```